pilot/pkg/model/jwks_resolver_test.go

```diff
@@ -317,6 +317,49 @@ func TestGetPublicKey(t *testing.T) {
 	}
 }
 
+func TestGetPublicKeyReorderedKey(t *testing.T) {
```

Comment:
I may add an unit test for `compareJWKSResponse` itself to cover all other corner cases of that function.

Reply:
Good idea, will add that in.